### PR TITLE
feat: behaviors, transitions, and conditions are reorderable

### PR DESCRIPTION
**Pull request by Maxsiom**
### Description
In the Step Inspector all behaviors, transitions, and conditions are reorderable now. They can be moved up and down. Therefore, a new 'ReorderableListOfAttribute' attribute is introduced.

**Fixes**: [TRNG-495](https://jira.innoactive.de/browse/TRNG-495)

### Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

### How Has This Been Tested?
This was tested manually, since it's a UI change.

**Test Configuration**: Home PC, Unity 2019.4.1f1 LTS

### Checklist

- [x] My code follows the [Coding Conventions](#coding-conventions)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation - **currently on it**
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: Runtime/Attributes/ReorderableListOfAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿using System;
+using System.Reflection;
+using System.Runtime.Serialization;
+using Innoactive.Creator.Core.Utils;
+using NUnit.Framework.Constraints;
```
Comment (tomwim) [inline Runtime/Attributes/ReorderableListOfAttribute.cs:5]: NUnit causes a compiler error and has to be removed. All usings except for System can be removed.

Comment (Gusinuhe) [inline Runtime/Attributes/ReorderableListOfAttribute.cs:5]: None of these are required.

File: Editor/UI/Drawers/MetadataWrapperDrawer.cs [hunk 2]
```diff
@@ -114,7 +179,7 @@ private Rect DrawDeletable(Rect rect, MetadataWrapper wrapper, Action<object> ch
                 fontStyle = FontStyle.Bold
             };
 
-            if (GUI.Button(new Rect(rect.x + rect.width - buttonSize.x, rect.y, buttonSize.x, buttonSize.y), "x", style))
+            if (GUI.Button(new Rect(rect.x + rect.width - buttonSize.x, rect.y, buttonSize.x, buttonSize.y), deleteIcon.Texture, style))
```
Comment (Gusinuhe) [inline Editor/UI/Drawers/MetadataWrapperDrawer.cs:182]: Awesome! 🙂

**Comment by tomwim** [comment]
Testing:
- Moving transitions up and down often does not work. Could not find a pattern why :/
- Cannot move behaviors within a Behavior Sequence
- Moving one transition onto a step can cause that multiple transitions are connected to that step

**Comment by Maxsiom** [comment]
Gray out buttons does work now but unfortunately is not updated when adding new behaviors and conditions. I am on a fix.

**Comment by Maxsiom** [comment]
Fixed gray out buttons.

Ready for rereview since I cannot reproduce any issues.

**Comment by tomwim** [comment]
image
Gray out buttons do not work for me

**Comment by tomwim** [comment]
Behaviors within sequence cannot be moved up and down. Not sure if intended
image

**Comment by Maxsiom** [comment]
> Behaviors within sequence cannot be moved up and down. Not sure if intended
> image

Cannot reproduce. Works like a charm for me. :(

behaviors

**Comment by Maxsiom** [comment]
Okay, found the issue: You need this small PR/Update. 👍 
https://github.com/Innoactive/Basic-Conditions-And-Behaviors/pull/22

**Comment by Maxsiom** [comment]
The transitions also work for me. Even your picture looks right to me. @tomwim 

transitions

**Pull request merged**